geopandas/geodataframe.py

```diff
@@ -20,7 +20,7 @@
 from ._decorator import doc
 
 
-DEFAULT_GEO_COLUMN_NAME = "geometry"
+DEFAULT_GEO_COLUMN_NAME = object()  # TODO this has a bad repr
```

Comment:
My main thinking was as a developer, I don't believe it will show up in error messages (unless I've done something wrong in this PR, because this should correspond to `.geometry` being unset, and so that error will trigger before the name could be seen.
You could see this as a user for example `GeoDataFrame()._geometry_column_name`.

This has made me wonder a bit more about other places it could show up, so I've been looking through usages of `_geometry_column_name`. I think `to_parquet` is the only case where it is access in an unguarded way which causes a problem (inside _create_metadata)
```
gpd.GeoDataFrame({"a":[1,2]}).to_parquet("foo.parquet")
...
--> 179     raise TypeError(f'Object of type {o.__class__.__name__} '
    180                     f'is not JSON serializable')

TypeError: Object of type object is not JSON serializable
```
This fails because of not having a repr I guess, but I think I should probably actually special case on DEFAULT_GEO_COLUMN_NAME and set the metadata geo col name to None in this case.